src/heap.c

```diff
@@ -81,7 +81,7 @@ typedef enum mi_collect_e {
 } mi_collect_t;
 
 
-static bool mi_heap_page_collect(mi_heap_t* heap, mi_page_queue_t* pq, mi_page_t* page, void* arg_collect, void* arg2 ) {
+static __attribute__((const)) bool mi_heap_page_collect(mi_heap_t* heap, mi_page_queue_t* pq, mi_page_t* page, const void* arg_collect, const void* arg2 ) {
```

Comment:
Ah we can't use `__attribute__((const))` as it is not standard C.  An option is to define a macro (MI_PURE) for this depending on the compiler but I don't think it is worth it -- just remove all the `__attribute__((const))`

Reply:
I removed __attribute __ ((const)) and rewrote the commit, sorry I'm not used to developing code outside of GNU systems that use GCC